src/OpenTelemetry.Api/Context/Propagation/BaggageFormat.cs

```diff
@@ -0,0 +1,145 @@
+{
+    /// <summary>
+    /// W3C baggage: https://github.com/w3c/baggage/blob/master/baggage/HTTP_HEADER_FORMAT.md.
+    /// </summary>
+    public class BaggageFormat : ITextFormat
+    {
+        private const string Baggage = "baggage";
+        private const int MaxBaggageLength = 1024;
+
+        /// <inheritdoc/>
+        public ISet<string> Fields => new HashSet<string> { Baggage };
+
+        /// <inheritdoc/>
+        public TextFormatContext Extract<T>(TextFormatContext context, T carrier, Func<T, string, IEnumerable<string>> getter)
+        {
+            if (carrier == null)
+            {
+                OpenTelemetryApiEventSource.Log.FailedToInjectActivityContext("null carrier");
+                return context;
+            }
+
+            if (getter == null)
+            {
+                OpenTelemetryApiEventSource.Log.FailedToExtractContext("null getter");
+                return context;
+            }
+
+            try
+            {
+                IEnumerable<KeyValuePair<string, string>> baggage = null;
+                var baggageCollection = getter(carrier, Baggage);
+                if (baggageCollection?.Any() ?? false)
+                {
+                    TryExtractTracestateBaggage(baggageCollection.ToArray(), out baggage);
+                }
+
+                return new TextFormatContext(
+                    context.ActivityContext,
+                    baggage ?? context.ActivityBaggage);
+            }
+            catch (Exception ex)
+            {
+                OpenTelemetryApiEventSource.Log.ActivityContextExtractException(ex);
+            }
+
+            return context;
+        }
+
+        /// <inheritdoc/>
+        public void Inject<T>(TextFormatContext context, T carrier, Action<T, string, string> setter)
+        {
+            if (carrier == null)
+            {
+                OpenTelemetryApiEventSource.Log.FailedToInjectActivityContext("null carrier");
+                return;
+            }
+
+            if (setter == null)
+            {
+                OpenTelemetryApiEventSource.Log.FailedToInjectActivityContext("null setter");
+                return;
+            }
+
+            using IEnumerator<KeyValuePair<string, string>> e = context.ActivityBaggage?.GetEnumerator();
+
+            if (e != null && e.MoveNext())
+            {
+                StringBuilder baggage = new StringBuilder();
+                do
+                {
+                    KeyValuePair<string, string> item = e.Current;
+                    baggage.Append(WebUtility.UrlEncode(item.Key)).Append('=').Append(WebUtility.UrlEncode(item.Value)).Append(',');
+                }
+                while (e.MoveNext());
+                baggage.Remove(baggage.Length - 1, 1);
+                setter(carrier, Baggage, baggage.ToString());
+            }
+        }
+
+        internal static bool TryExtractTracestateBaggage(string[] baggageCollection, out IEnumerable<KeyValuePair<string, string>> baggage)
```

Comment:
I think we've aligned to use `TraceState` instead of `Tracestate`?

Reply:
It should really just be `TryExtractBaggage` that was probably a copy/paste fail. I'll update later unless @eddynaka gets to it first.

Reply:
just updated.